packages/web-app-files/src/components/SideBar/Shares/FileShares.vue

```diff
@@ -227,27 +238,44 @@ export default defineComponent({
       }
       return this.resource.canShare({ user: this.user })
     },
+
     noResharePermsMessage() {
       const translatedFile = this.$gettext("You don't have permission to share this file.")
       const translatedFolder = this.$gettext("You don't have permission to share this folder.")
       return this.resource.type === 'file' ? translatedFile : translatedFolder
     },
+
     showSpaceMembers() {
       return (
         this.space?.driveType === 'project' &&
         this.resource.type !== 'space' &&
         this.currentUserIsMemberOfSpace
       )
     },
+
     matchingSpace() {
       return this.space || this.spaces.find((space) => space.id === this.resource.storageId)
     }
   },
   methods: {
-    ...mapActions('Files', ['deleteShare']),
+    ...mapActions('Files', ['deleteShare', 'addShare']),
     ...mapActions(['createModal', 'hideModal', 'showMessage']),
     ...mapMutations('Files', ['REMOVE_FILES']),
 
+    getDeniedShare(collaborator) {
```

Comment:
Would love to get some types in these methods here, I always wonder how `collaborator` looks like. I think it should be of type `Share`.